src/refdb_fs.c

```diff
@@ -63,6 +66,18 @@ typedef struct refdb_fs_backend {
 	uint32_t direach_flags;
 } refdb_fs_backend;
 
+/*
+ * List of symbolic reference names that are always stored in the
+ * gitdir. This is required such that we are able to correctly
+ * resolve these references via the gitdir instead of the
+ * commondir when working trees are used.
+ */
+static const char *symrefs[] = {
```

Comment:
I suspect that we should give this a more meaningful name.  (And the `is_symbolic_ref` function, also, below.)  You could have a symbolic ref `refs/heads/foo` -> `refs/heads/bar`, which would be in the commondir not the gitdir.

I don't have any brilliant ideas though.  Special refs?  Local refs?  Workdir refs?